src/npctalk_rules.cpp

```diff
@@ -0,0 +1,518 @@
+        guy->rules = default_values;
+    }
+
+    int rule_number = 0;
+    /* Handle all of our regular, boolean rules */
+    for( const std::pair<const std::string, ally_rule_data> &rule_data : ally_rule_strs ) {
+        ImGui::PushID( rule_number );
+        assigned_hotkey = input_ptr->next_unassigned_hotkey( hotkeys, assigned_hotkey );
+        rule_number++;
+        ImGui::NewLine();
+        print_hotkey( assigned_hotkey );
+        const ally_rule_data &this_rule = rule_data.second;
+        bool rule_enabled = false;
+        std::string rules_text;
+        if( guy->rules.has_flag( this_rule.rule ) ) {
+            rules_text = string_format( "%s", get_parsed( this_rule.rule_true_text ) );
+            rule_enabled = true;
+        } else {
+            rules_text = string_format( "%s", get_parsed( this_rule.rule_false_text ) );
+        }
+        if( rule_enabled ) {
+            ImGui::PushStyleColor( ImGuiCol_Button, c_green );
+        }
+        if( ImGui::Button( _( "Toggle" ) ) || pressed_key == assigned_hotkey ) {
+            ImGui::SetKeyboardFocusHere( -1 );
+            guy->rules.toggle_flag( this_rule.rule );
+            guy->rules.toggle_specific_override_state( this_rule.rule, !rule_enabled );
+        }
+        ImGui::PopStyleColor();
+        ImGui::SameLine();
+        if( ImGui::Button( _( "Default" ) ) ) {
+            guy->rules.clear_flag( this_rule.rule );
+            guy->rules.clear_override( this_rule.rule );
+        }
+        ImGui::SameLine();
+        draw_colored_text( rules_text );
+        ImGui::PopID();
+    }
+
+    // Engagement rules require their own set of buttons, each instruction is unique
+    ImGui::Separator();
+    ImGui::NewLine();
+    assigned_hotkey = input_ptr->next_unassigned_hotkey( hotkeys, assigned_hotkey );
+    print_hotkey( assigned_hotkey );
+    multi_rule_header( "ENGAGEMENT_RULES", guy->rules.engagement, engagement_rules,
+                       pressed_key == assigned_hotkey );
+    int engagement_rule_number = 0;
+    for( const std::pair<const combat_engagement, std::string> &engagement_rule : engagement_rules ) {
+        ImGui::PushID( rule_number );
+        rule_number++;
+        engagement_rule_number++;
+        // Could use a better label for these...
+        std::string button_label = std::to_string( engagement_rule_number );
+        ImGui::SameLine();
+        if( guy->rules.engagement == engagement_rule.first ) {
+            ImGui::PushStyleColor( ImGuiCol_Button, c_green );
+        }
+        if( ImGui::Button( button_label.c_str() ) ) {
+            guy->rules.engagement = engagement_rule.first;
+            ImGui::SetKeyboardFocusHere( -1 );
+        }
+        ImGui::PopStyleColor();
+        ImGui::PopID();
+    }
+    ImGui::SameLine();
+    draw_colored_text( _( "Engagement rules" ), c_white );
+    ImGui::NewLine();
+    draw_colored_text( string_format( "%s", get_parsed( engagement_rules[guy->rules.engagement] ) ) );
```

Comment:
This business of looping over a list of rules is rather repetitive. Pull it out into a function? Call it `rule_list` or something? I see that for `ally_rules` the body is somewhat different though, so maybe some thought is required.